scipy/interpolate/fitpack/fpknot.f

```diff
@@ -20,7 +20,11 @@ subroutine fpknot(x,m,t,n,fpint,nrdata,nrint,nest,istart)
       real*8 an,am,fpmax
       integer ihalf,j,jbegin,jj,jk,jpoint,k,maxbeg,maxpt,
      * next,nrx,number
-c  ..
+
+c  note: do not initialize on the same line to avoid saving between calls
```

Comment:
Interesting that this matters - is this a known Fortran gotcha?

Reply:
not a fortran expert, but see "GOTCHA - initialisation implies variable is" in https://learnxinyminutes.com/docs/fortran95/

Reply:
As a data point, I did not know that. I suspect it's fixed by the `recursive subrourune` update. But this is obscure and surely implementation dependent. "Implemenration-defined" in C-speak.

Reply:
For me (not a fortran/C interop expert, just pure fortran experience) a big surprise is that debug prints change the codegen. I used to think that it's a C UB thing, and in Fortran debug prints are safe.

Reply:
to follow up on ralf's question: yes, this seems to be a common gotcha. there are tons of threads on this and proposals to mitigate this in various implementations. e.g., https://comp.lang.fortran.narkive.com/JEaAXM9c/warn-about-implicit-save

as for the codegen issue, i think it might be similar to what people might experience with c. lots of "unexpected" optimizations are usually due to UB, such as relying on uninitialized data (that compiler sees as "it's okay to optimize this"), but might not be what people expect.